protoc-plugin/src/main/java/io/spine/tools/protoc/SpineProtoGenerator.java

```diff
@@ -115,33 +115,51 @@ protected SpineProtoGenerator() {
      *
      * @param request the compiler request
      * @return the response to the compiler
-     * @see #processMessage(FileDescriptorProto, DescriptorProto) for more detaineed behavior
-     *                                                            description
+     * @see #processMessage(FileDescriptorProto, DescriptorProto) for more detailed description
      */
     public final CodeGeneratorResponse process(CodeGeneratorRequest request) {
         checkNotNull(request);
-        checkNotNull(request);
-        final Version protocVersion = request.getCompilerVersion();
-        checkArgument(protocVersion.getMajor() >= 3,
-                      "Use protoc of version 3.X.X or higher to generate the Spine sources.");
-        final List<FileDescriptorProto> descriptors = request.getProtoFileList();
-        checkArgument(!descriptors.isEmpty(), "No files to generate provided.");
-        final CodeGeneratorResponse response = scan(descriptors);
+        checkCompilerVersion(request);
+        final List<FileDescriptorProto> protoFiles = request.getProtoFileList();
+        checkArgument(!protoFiles.isEmpty(), "No files to generate provided.");
+        final CodeGeneratorResponse response = process(protoFiles);
         return response;
     }
 
-    private CodeGeneratorResponse scan(Iterable<FileDescriptorProto> files) {
+    /**
+     * Ensures that the version of the Google Protobuf Compiler is 3.* or higher.
```

Comment:
AFAIK `3.0-beta1` will pass, but we cannot work with it.

Reply:
The code currently checks for `>= 3`. Let's keep it the way it is, at least for this PR. We can make the check stricter, when we observe incompatible usage reports.